examples/rust/examples/maker/maker.rs

```diff
@@ -38,100 +41,117 @@ const TOS_ACCEPTANCE: &str = "I accept the Valorem Terms of Service at https://a
+                // trying to implement reconnection logic as most of the time the node will be local.
+                panic!("HTTP connection error: {connection_error:?}")
+            }
         };
 
-        op.retry(
-            &ExponentialBuilder::default()
-                .with_jitter()
-                .with_max_times(10),
-        )
-        .await?;
-    } else if settings.node_endpoint.starts_with("ws") {
-        // Websockets (ws & wss)
-        let op = || async {
-            warn!("Re/starting maker");
-            run(
-                Arc::new(Provider::<Ws>::new(
-                    Ws::connect(settings.node_endpoint.clone()).await?,
-                )),
-                settings.clone(),
-            )
-            .await
+        EthersProvider::HttpProvider(provider)
+    } else if node_endpoint.starts_with("ws") {
+        let provider = match Ws::connect(node_endpoint).await {
+            Ok(connection) => {
+                // Decrease the ethers polling time from the default of 7 seconds to 1 second.
+                Provider::<Ws>::new(connection).interval(Duration::from_secs(1))
+            }
+            Err(connection_error) => {
+                // If we cannot connect to our provider, we are in trouble. Panic out instead of
+                // trying to implement reconnection logic as most of the time the node will be local.
+                panic!("Websocket connection error: {connection_error:?}")
+            }
         };
 
-        op.retry(
-            &ExponentialBuilder::default()
-                .with_jitter()
-                .with_max_times(10),
-        )
-        .await?;
+        EthersProvider::WsProvider(provider)
     } else {
-        // IPC
-        let op = || async {
-            warn!("Re/starting maker");
-            run(
-                Arc::new(Provider::connect_ipc(settings.node_endpoint.clone()).await?),
-                settings.clone(),
-            )
-            .await
+        let provider = match Provider::connect_ipc(node_endpoint).await {
+            Ok(connection) => {
+                // Decrease the ethers polling time from the default of 7 seconds to 1 second.
+                connection.interval(Duration::from_secs(1))
+            }
+            Err(connection_error) => {
+                // If we cannot connect to our local provider, we are in trouble just panic out instead.
+                panic!("IPC connection error: {connection_error:?}")
+            }
         };
 
-        op.retry(
-            &ExponentialBuilder::default()
-                .with_jitter()
-                .with_max_times(10),
-        )
-        .await?;
+        EthersProvider::IpcProvider(provider)
     }
-
-    Ok(())
 }
 
 // Main execution function. This is not expected to return.
 async fn run<P: JsonRpcClient + 'static>(
     provider: Arc<Provider<P>>,
     settings: Settings,
-) -> Result<(), anyhow::Error> {
+) -> Option<()> {
```

Comment:
losing error info here which would be useful for debugging. how do we no it's not continually hitting an error and reconnecting?

Reply:
i would probably keep as result, propagate the errors with `?`s, then log the error in the reconnect loop

Reply:
> how do we no it's not continually hitting an error and reconnecting?
The main loop will be logging it was forced to reconnect.

Added additional context around reconnection events.